contracts/Vesting.sol

```diff
@@ -1,6 +1,131 @@
 // SPDX-License-Identifier: MIT
 
-pragma solidity 0.8.14;
+pragma solidity 0.8.7;
 
-contract Vesting {
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "../interfaces/IVesting.sol";
+
+contract Vesting is IVesting, Ownable {
+    using SafeERC20 for IERC20;
+
+    struct Beneficiary {
+        uint256 intialReward;
+        uint256 rewardPaid;
+        uint256 balanceBase;
+    }
+
+    mapping(address => Beneficiary) public listOfBeneficiaries;
+    mapping(Allocation => mapping(address => Beneficiary))
+        public allocationList;
+    uint256 public vestingStartDate;
+    uint256 public vestingDuration;
+    uint256 public vestingCliff;
+
+    IERC20 private _token;
+
+    constructor(address token_) {
+        require(
+            Address.isContract(token_),
+            "Error : Incorrect address , only contract address"
+        );
+        _token = IERC20(token_);
+    }
+
+    function setInitialTimestamp(uint256 initialTimestamp_)
+        external
+        override
+        onlyOwner
+    {
+        require(
+            initialTimestamp_ != 0,
+            "Error : 'initialTimestamp_' must be greater than 0"
+        );
+        require(vestingStartDate == 0, "error : can call only once time");
+        vestingStartDate = block.timestamp + initialTimestamp_;
+        vestingCliff = vestingStartDate + 10 minutes;
+        vestingDuration = vestingCliff + 600 minutes;
+        emit VestingStart(vestingStartDate);
+    }
+
+    function addInvestors(
+        address[] calldata investors_,
+        uint256[] calldata amount_,
+        Allocation[] calldata allocation_
+    ) external override onlyOwner {
+        require(
+            investors_.length == amount_.length &&
+                investors_.length == allocation_.length,
+            "Error : Different arrays length"
+        );
+        uint256 sumOfAmount;
+        for (uint256 i; i < investors_.length; i++) {
+            require(
+                investors_[i] > address(0) && amount_[i] > 0,
+                "Error : 'investors_' or 'amount_' , equal to 0"
+            );
+            if (allocation_[i] == Allocation.Seed) {
+                allocationList[allocation_[i]][
+                    investors_[i]
+                ] = listOfBeneficiaries[investors_[i]];
+                listOfBeneficiaries[investors_[i]].intialReward += ((amount_[
+                    i
+                ] / 10000) * 1000);
+                listOfBeneficiaries[investors_[i]].balanceBase += amount_[i];
+            }
+            if (allocation_[i] == Allocation.Private) {
```

Comment:
if else constructions is redundant, think about how it could be done to get rid of checks for the allocation type